src/charm.py

```diff
@@ -14,74 +16,72 @@
+        return self._get_unit_hostname(unit_number)
 
     @property
-    def hostnames(self) -> list:
-        """Return a list of Kubernetes hostnames."""
-        return [self._get_unit_hostname(i) for i in range(self.cluster_size)]
+    def env_config(self) -> dict:
+        """Return the env_config for the Kubernetes pod_spec"""
+        config = self.model.config
+        env_config = {}
 
-    def _on_start(self, event):
-        """Initialize MySQL InnoDB cluster.
+        if config.get("MYSQL_ROOT_PASSWORD"):
+            env_config["MYSQL_ROOT_PASSWORD"] = config["MYSQL_ROOT_PASSWORD"]
+        else:
+            env_config[
+                "MYSQL_ROOT_PASSWORD"
+            ] = self._stored.MYSQL_ROOT_PASSWORD
+            logger.warning(
+                "The randomly generated MYSQL_ROOT_PASSWORD is: %s",
+                self._stored.MYSQL_ROOT_PASSWORD,
+            )
+            logger.warning("Please change it as soon as possible!")
 
-        This event handler is deferred if initialization of MySQL
-        replica set fails. By doing so it is guaranteed that another
-        attempt at initialization will be made.
-        """
-        if not self.unit.is_leader():
-            return
+        if config.get("MYSQL_USER") and config.get("MYSQL_PASSWORD"):
+            env_config["MYSQL_USER"] = config["MYSQL_USER"]
+            env_config["MYSQL_PASSWORD"] = config["MYSQL_PASSWORD"]
+
+        if config.get("MYSQL_DATABASE"):
+            env_config["MYSQL_DATABASE"] = config["MYSQL_DATABASE"]
+
+        return env_config
+
+    def _on_start(self, event):
+        """Initialize MySQL"""
 
         if not self._mysql_is_ready():
             message = "Waiting for MySQL Service"
             self.unit.status = WaitingStatus(message)
             logger.info(message)
             event.defer()
             return
-        else:
-            self.unit.status = ActiveStatus("MySQL Service is ready")
-
-        # create admin users on all hosts
-        for hostname in self.hostnames:
-            self._create_idcadmin_user_on_host(hostname)
-
-    def _create_idcadmin_user_on_host(self, hostname):
-        """This method will execute a user creation query for the idcAdmin user."""
-
-        if len(self.model.config['idcAdmin-password']) == 0:
-            message = 'idcAdmin-password not provided'
-            logger.error(message)
-            self.unit.status = BlockedStatus(message)
-            return
-
-        logger.warning('Creating user on {}'.format(hostname))
-        query = """
-                CREATE USER 'idcAdmin'@'%' IDENTIFIED BY '{}';
-                GRANT ALL ON *.* TO 'idcAdmin'@'%' WITH GRANT OPTION";
-                """.format(self.model.config['idcAdmin-password'])
 
-        cnx = self._get_sql_connection_for_host(hostname)
-        cur = cnx.cursor()
-        cur.execute(query)
-        cnx.close()
+        self.unit.status = ActiveStatus("MySQL is up and running!")
```

Comment:
I believe that it is recommended to not set a message for `ActiveStatus` except in rare cases.

Reply:
ack